src/python/gudhi/nerve_gic.pyx

```diff
@@ -400,23 +408,726 @@ cdef class CoverComplex:
+                interval_endpoints, l = np.linspace(min_f, max_f, num=r+1, retstep=True)
+                for j in range(len(interval_endpoints)-1):
+                    L.append(interval_endpoints[j]   - g*l / (2 - 2*g))
+                    R.append(interval_endpoints[j+1] + g*l / (2 - 2*g))
+                l_int.append(L)
+                r_int.append(R)
+
+            # Build the binned_data map that takes a patch or a patch intersection and outputs the indices of the points contained in it
+            binned_data = {}
+            for i in range(num_pts):
+                list_preimage = []
+                for j in range(num_filters):
+                    fval = self.filters[i,j]
+                    start, end = int(min(np.argwhere(np.array(r_int[j]) >= fval))), int(max(np.argwhere(np.array(l_int[j]) <= fval)))
+                    list_preimage.append(list(range(start, end+1)))
+                list_preimage = list(itertools.product(*list_preimage))
+                for pre_idx in list_preimage:
+                    try:
+                        binned_data[pre_idx].append(i)
+                    except KeyError:
+                        binned_data[pre_idx] = [i]
+
+        # Initialize the cover map, that takes a point and outputs the clusters to which it belongs
+        cover, clus_base = [[] for _ in range(num_pts)], 0
+
+        # For each patch
+        for preimage in binned_data:
+
+            # Apply clustering on the corresponding subpopulation
+            idxs = np.array(binned_data[preimage])
+            if len(idxs) > 1:
+                clusters = self.clustering.fit_predict(X[idxs,:]) if self.input == "point cloud" else self.clustering.fit_predict(X[idxs,:][:,idxs])
+            elif len(idxs) == 1:
+                clusters = np.array([0])
+            else:
+                continue
+
+            # Collect various information on each cluster
+            num_clus_pre = np.max(clusters) + 1
+            for clus_i in range(num_clus_pre):
+                node_name = clus_base + clus_i
+                subpopulation = idxs[clusters == clus_i]
+                self.node_info_[node_name] = {}
+                self.node_info_[node_name]["indices"] = subpopulation
+                self.node_info_[node_name]["size"] = len(subpopulation)
+                self.node_info_[node_name]["colors"] = np.mean(self.colors[subpopulation,:], axis=0)
+                self.node_info_[node_name]["patch"] = preimage
+
+            # Update the cover map
+            for pt in range(clusters.shape[0]):
+                node_name = clus_base + clusters[pt]
+                if clusters[pt] != -1 and self.node_info_[node_name]["size"] >= self.mask:
+                    cover[idxs[pt]].append(node_name)
+
+            clus_base += np.max(clusters) + 1
+
+        # Insert the simplices of the Mapper complex 
+        for i in range(num_pts):
+            self.mapper_.insert(cover[i], filtration=-3)
+        self.mapper_.initialize_filtration()
```

Comment:
No more need to call `initialize_filtration()`.
```txt
DeprecationWarning: Since Gudhi 3.2, calling SimplexTree.initialize_filtration is unnecessary.
```

Reply:
Thanks! Done